src/engine/game.ts

```diff
@@ -51,34 +57,87 @@ export default class Game {
 
   stop(): void {
     this.running = false;
-    TimeService.currentFrame = undefined;
+    this.currentFrame = 0;
+
+    Game.inputControllers.forEach(e => e.unbind());
   }
 
   add(gameObject: GameObject): void {
     this.gameObjects.push(gameObject);
   }
 
+  addInput(inputController: InputController): void {
+    Game.inputControllers.push(inputController);
+    if (this.running) inputController.bind();
+  } 
+
   removeAllGameObjects(): void {
     this.gameObjects = [];
   }
 
   redraw(): void {
-    RenderEngine.render(this.canvas, this.gameObjects);
+    RenderEngine.render(this.context, this.gameObjects);
   }
 
-  update(timestamp: number | undefined = undefined): void {
-    if (timestamp === undefined || TimeService.currentFrame === undefined) {
-      TimeService.delta = 0;
-    } else {
-      TimeService.delta = timestamp - TimeService.currentFrame;
+  update(timestamp: number = 0): void {
+
+    // Time update
+    var delta = 0;
+    if (timestamp > 0 && this.currentFrame > 0) {
+      delta = timestamp - this.currentFrame;
     }
+    this.currentFrame = timestamp;
 
-    this.gameObjects.forEach(e => e.update(this.gameInfo));
-    this.redraw();
-    TimeService.currentFrame = timestamp;
+    // Remove pass
+    for (var i = this.gameObjects.length - 1; i >= 0; i--) {
+      if (!this.gameObjects[i].alive) {
+        this.gameObjects.splice(i, 1);
+      }
+    } 
+
+    // Event pass
+    while (Game.events.length > 0) {
+      EventSystem.handleEvent(this, Game.events.shift());
+    }
+    
+    // Udpate pass
+    this.gameObjects.forEach(e => e.update(delta));
 
+    // Draw pass
+    this.redraw();
+    
+    // Loop
     if (this.running) {
       window.requestAnimationFrame((t) => this.update(t));
     }
   }
-};
\ No newline at end of file
+
+  private static _viewPortWidth: number = 0;
+  private static _viewPortHeight: number = 0;
+
+  static Info = class {
+
+    static get viewPortHeight(): number {
+      return Game._viewPortHeight;
+    }
+
+    static get viewPortWidth(): number {
+      return Game._viewPortWidth;
+    }
+  }
```

Comment:
Yeah I was kinda thinking about that idea too. We probably only ever want one game, but that doesn't mean it should break if somebody makes 2